OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -0,0 +1,457 @@
+    return 0;
+}
+
+//=============================================================================
+// Test Cases
+//=============================================================================
+
+void testAccuracy()
+{
+    cout << "\ntestInverseKinematicsSolver::testAccuracy()" << endl;
+
+    std::unique_ptr<Model> pendulum{ constructPendulumWithMarkers() };
+    Coordinate& coord = pendulum->getCoordinateSet()[0];
+
+    double refVal = 0.123456789;
+    double looseAccuracy = 1e-3;
+    double tightAccuracy = 1.0e-9;
+
+    SimTK::Array_<CoordinateReference> coordRefs;
+    Constant coordRefFunc(refVal);
+    CoordinateReference coordRef(coord.getName(), coordRefFunc);
+    coordRef.setWeight(1.0);
+    coordRefs.push_back(coordRef);
+
+    SimTK::State state = pendulum->initSystem();
+    double coordValue = coord.getValue(state);
+    double refVal0 = coordRefs[0].getValue(state);
+
+    cout.precision(10);
+    cout << "Initial " << coord.getName() << " value = " << coordValue <<
+        " referenceValue = " << refVal0 << endl;
+
+    coord.setValue(state, refVal);
+    StatesTrajectory states;
+    states.append(state);
+
+    MarkerData* markerData =
+        generateMarkerDataFromModelAndStates(*pendulum, states);
+    // MarkersReference takes ownership of the markerData
+    MarkersReference markersRef(markerData);
+    markersRef.setDefaultWeight(1.0);
+
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    InverseKinematicsSolver ikSolver(*pendulum, markersRef, coordRefs);
+    ikSolver.setAccuracy(looseAccuracy);
+    ikSolver.assemble(state);
+
+    coordValue = coord.getValue(state);
+    cout << "Assembled " << coord.getName() << " value = " << coordValue << endl;
+
+    double accuracy = abs(coordValue - refVal);
+    cout << "Specified accuracy: " << looseAccuracy << "; achieved: "
+        << accuracy << endl;
+
+    // test should fail if we did not meet the target accuracy
+    SimTK_ASSERT_ALWAYS(accuracy <= looseAccuracy,
+        "InverseKinematicsSolver failed to meet specified accuracy");
+    
+    SimTK::Array_<double> sqMarkerErrors;
+    double looseSumSqError = 0;
+    ikSolver.computeCurrentSquaredMarkerErrors(sqMarkerErrors);
+    for (const double& err : sqMarkerErrors) {
+        looseSumSqError += err;
+    }
+    
+    cout << "For accuracy: " << looseAccuracy << "; Sum-squared Error: " 
+        << looseSumSqError << endl;
+
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    ikSolver.setAccuracy(tightAccuracy);
+    ikSolver.assemble(state);
+
+    coordValue = coord.getValue(state);
+    cout << "Assembled " << coord.getName() <<" value = "<< coordValue << endl;
+
+    accuracy = abs(coord.getValue(state) - refVal);
+
+    cout << "Specified accuracy: " << tightAccuracy << "; achieved: "
+        << accuracy << endl;
+
+    // test should fail if we did not meet the new tighter target accuracy
+    SimTK_ASSERT_ALWAYS(accuracy <= tightAccuracy,
```

Comment:
As above: I misinterpreted this as if it were `OPENSIM_THROW_IF()` , I think because the comment describes the conditions for failure but the assert states conditions for success. No changes necessary.